### PR TITLE
Add profile data to provisioning api stable11

### PR DESCRIPTION
**Pull request by schiessle**
backport of https://github.com/nextcloud/server/pull/3232

### CODE REVIEW

File: apps/provisioning_api/lib/Controller/UsersController.php [hunk 1]
```diff
@@ -209,17 +215,48 @@ public function getUser($userId) {
...
+	 * @return DataResponse
+	 * @throws OCSException
+	 */
+	public function getCurrentUser() {
+		$user = $this->userSession->getUser();
+		if ($user) {
+			$result =  $this->getUser($user->getUID());
+			// rename "displayname" to "display-name" only for this call to keep
+			// the API stable.
+			$result['display-name'] = $result['displayname'];
```
Comment (MorrisJobke) [inline apps/provisioning_api/lib/Controller/UsersController.php]:
```
{"reqId":"KRAJ10EAy8L+Su0X2vz7","remoteAddr":"::1","app":"PHP","message":"Cannot use object of type OCP\\AppFramework\\Http\\DataResponse as array at \/Users\/morrisjobke\/Projects\/nextcloud\/server\/apps\/provisioning_api\/lib\/Controller\/UsersController.php#248","level":3,"time":"2017-01-26T03:26:07+00:00","method":"GET","url":"\/server\/ocs\/v1.php\/cloud\/user","user":"admin","version":"11.0.1.2"}
```

**Review by MorrisJobke — CHANGES_REQUESTED** [review]

File: apps/provisioning_api/lib/Controller/UsersController.php (hunk 1 above)
Comment (MorrisJobke) [inline apps/provisioning_api/lib/Controller/UsersController.php]: Same on `master`, but I'm pretty sure, that I tested this in the other PR review 😕 

@schiessle Could you fix this on master and then cherry-pick the fix to this PR. Thanks :)

Comment (schiessle) [inline apps/provisioning_api/lib/Controller/UsersController.php]: yes, there is even a unit test which should fail in this case... Will have a look.

**Comment by schiessle** [comment]
@MorrisJobke last commit should fix it

**Review by MorrisJobke — APPROVED** [review]
Tested and works 👍

**Review by LukasReschke — APPROVED** [review]

**Pull request merged**